lib/publiccloud/instance.pm

```diff
@@ -66,7 +66,7 @@ sub scp {
     $from =~ s/^remote:/$url/;
     $to   =~ s/^remote:/$url/;
 
-    my $ssh_cmd = sprintf('scp %s -i "%s" "%s" "%s"',
+    my $ssh_cmd = sprintf('scp %s -i "%s" %s %s',
```

Comment:
I don't understand this. What about filenames with spaces, I would avoid letting the caller take care of the escaping.
Can you show me a run where you had problems with it?

Reply:
http://fromm.arch.suse.de/tests/6474/file/serial_terminal.txt

> # scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "/root/.ssh/id_rsa" "~/ltp-20190702.a244bcc22-qa.2.1.x86_64.rpm" "susetest@34.77.110.200:~/ltp-20190702.a244bcc22-qa.2.1.x86_64.rpm" ; echo jVxuv-$?-
> ~/ltp-20190702.a244bcc22-qa.2.1.x86_64.rpm: No such file or directory

Reply:
I see, the problem is that the file path you given contains `~/` and this isn't interpreted anymore. The `~/` is handled by local bash, so you could also write `$HOME/`.
But I prefer using a absolute path.

Reply:
ok, I leave it intact and use absolute path